api/admin/controller.py

```diff
@@ -987,59 +992,16 @@ def libraries(self):
-        
-        protocols.append({
-            "name": ExternalIntegration.OPDS_IMPORT,
-            "fields": [
-                { "key": "external_account_id", "label": _("URL") },
-            ],
-        })
-
-        protocols.append({
-            "name": ExternalIntegration.OVERDRIVE,
-            "fields": [
-                { "key": "external_account_id", "label": _("Library ID") },
-                { "key": "website_id", "label": _("Website ID") },
-                { "key": ExternalIntegration.USERNAME, "label": _("Client Key") },
-                { "key": ExternalIntegration.PASSWORD, "label": _("Client Secret") },
-                { "key": "default_loan_period", "label": _("Default Loan Period") },
-                { "key": "default_reservation_period", "label": _("Default Reservation Period") },
-            ],
-        })
-
-        protocols.append({
-            "name": ExternalIntegration.BIBLIOTHECA,
-            "fields": [
-                { "key": ExternalIntegration.USERNAME, "label": _("Account ID") },
-                { "key": ExternalIntegration.PASSWORD, "label": _("Account Key") },
-                { "key": "external_account_id", "label": _("Library ID") },
-                { "key": "default_loan_period", "label": _("Default Loan Period") },
-                { "key": "default_reservation_period", "label": _("Default Reservation Period") },
-            ],
-        })
-
-        protocols.append({
-            "name": ExternalIntegration.AXIS_360,
-            "fields": [
-                { "key": ExternalIntegration.USERNAME, "label": _("Username") },
-                { "key": ExternalIntegration.PASSWORD, "label": _("Password") },
-                { "key": "external_account_id", "label": _("Library ID") },
-                { "key": ExternalIntegration.URL, "label": _("Server") },
-                { "key": "default_loan_period", "label": _("Default Loan Period") },
-                { "key": "default_reservation_period", "label": _("Default Reservation Period") },
-            ],
-        })
-
-        protocols.append({
-            "name": ExternalIntegration.ONE_CLICK,
-            "fields": [
-                { "key": ExternalIntegration.PASSWORD, "label": _("Basic Token") },
-                { "key": "external_account_id", "label": _("Library ID") },
-                { "key": ExternalIntegration.URL, "label": _("URL") },
-                { "key": "default_loan_period", "label": _("Default Loan Period") },
-                { "key": "default_reservation_period", "label": _("Default Reservation Period") },
-            ],
-        })
+        for license_api in [OPDSImporter,
+                            OverdriveAPI,
+                            BibliothecaAPI,
+                            Axis360API,
+                            OneClickAPI,
+                           ]:
+            protocols.append({
+                "name": license_api.NAME,
+                "fields": license_api.FIELDS,
+            })
```

Comment:
It's hard to say so early how many fields there are going to be to inherit. At any rate, either way we'll have to make this change with every new content source. I'm fine with either solution.